ssh/ssh_tunnel.py

```diff
@@ -0,0 +1,138 @@
+"""
+Module for creating persistent SSH connections for use with synchronous
+commands. Typically, tunnels should be invoked with a context manager to
+ensure proper cleanup. E.G.:
+with contextlib.closing(SSHTunnel(*args, **kwargs)) as tunnel:
+    ssh.write_to_remote('/usr/local/usrpath/testfile.txt', 'test_file.txt')
+    ssh.remote_cmd(['cat', 'test_file.txt'])
+"""
+import logging
+from contextlib import closing
+from subprocess import CalledProcessError, TimeoutExpired, check_call, check_output
+
+logger = logging.getLogger(__name__)
+
+SOCKET_DIR = '$HOME/.cache/ssh'
+SSH_CMD = [
+    '/usr/bin/ssh',
+    '-oConnectTimeout=10',
+    '-oControlMaster=auto',
```

Comment:
@kartick How about trapping or using context manager to cleanup sockets on specified signals?